lib/ansible/plugins/action/net_put.py

```diff
@@ -148,7 +149,11 @@ def _handle_existing_file(self, conn, source, dest, proto, timeout):
                 proto=proto, timeout=timeout
             )
         except Exception as exc:
-            if (to_text(exc)).find("No such file or directory") > 0:
+            pattern = to_text(exc)
+            not_found_exc = "No such file or directory"
+            if re.search(not_found_exc, pattern, re.I):
```

Comment:
Note that this is slower than the old code.  The standard way to do it would have been:

``` python
if u'no such file or directory' in to_text(exc).lower():
    if os.path.exists(source_file):
        os.remove(source_file)
    return True
```